dynamite_nsm/services/kibana/package/manager.py

```diff
@@ -0,0 +1,784 @@
+        package = Package(manifest, instobjs, id=pkg.get('id'))
+        return package
+
+
+    @staticmethod
+    def find_by_slug(package_slug: str):
+        """Find a package by its slug
+
+        Args:
+            package_slug (str): slug for package
+
+        Raises:
+            ValueError: Something went wrong performing search. e.g: ES is down
+
+        """
+        query = {
+            "query": {
+                "match":{
+                    "manifest.slug":{
+                        "query": package_slug,
+                        "minimum_should_match": 1
+                    }
+                }
+            }
+        }
+        
+        result = Package.es_search(query)
+        if not result:
+            return None
+        packagesdata = [r['_source'] for r in result['hits']['hits']]
+        pkg = packagesdata[0]
+        manifest = PackageManifest(pkg.get('manifest'))
+        instobjs = [InstalledObject.from_kwargs(**iobj) for iobj in pkg.get('installed_objects')]
+        package = Package(manifest, instobjs, id=pkg.get('id'))
+        return package
+
+    @staticmethod
+    def search_installed_packages(package_name=None) -> list:
+        """
+        Returns Packages with wildcard search on provided package name string,
+            returns all packages if package name not supplied
+
+        Args:
+            package_name ([str], optional): UUID of the package. Defaults to None.
+
+
+        Returns:
+            list: List of Packages
+        """
+        query = {
+            "aggs":{
+                "manifest.name": {
+                    "terms": {
+                        "field": "manifest.name.keyword"
+                    }
+                }
+            }
+        }
+
+        if package_name:
+            query["query"] = {
+                "wildcard":{
+                    "package_name":{
+                        "wildcard": f"*{package_name}*",
+                    }
+                }
+            }
+
+        result = Package.es_search(query)
+        if not result:
+            return None
+        
+        # If we want to just display the titles and num packages, we can pull from aggs result instead.
+        packagesdata = [r['_source'] for r in result['hits']['hits']]
+        packages = []
+        for pkg in packagesdata:
+            manifest = PackageManifest(pkg.get('manifest'))
+            instobjs = [InstalledObject.from_kwargs(**iobj) for iobj in pkg.get('installed_objects')]
+            package = Package(manifest, instobjs, id=pkg.get('id'))
+            packages.append(package)
+        return packages
+
+class SavedObjectsManager(object):
+    def __init__(self,
+                 stdout: Optional[bool] = True,
+                 verbose: Optional[bool] = False):
+        """Initializes the SavedObjectsManager
+
+        Args:
+            stdout (Optional[bool], optional): [description]. Defaults to True.
```

Comment:
Please use the following for stdout/verbose to maintain consistency across other command-line utilities.       
```      
stdout: Print the output to console
verbose: Include detailed debug messages
```